From b74c919c43dc7504e3031340728eb2cac9c94d58 Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Tue, 26 Sep 2017 12:25:09 -0700
Subject: [PATCH] PUBAPI-1445 improve CreateMachine affinity error message if a
 "==" affinity matches no VMs Reviewed by: Justin Reagor <justinwr@gmail.com>

---
 lib/triton-affinity.js | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/lib/triton-affinity.js b/lib/triton-affinity.js
index e41fd4e..43478e9 100644
--- a/lib/triton-affinity.js
+++ b/lib/triton-affinity.js
@@ -956,6 +956,12 @@ function localityFromRulesInfo(opts) {
     });
     var nearServerUuids = null; // Servers that satisfy all near rules.
     nearRules.forEach(function (nearRule) {
+        if (nearRule.vms.length === 0) {
+            throw new VError('cannot satisfy affinity rule "%s", '
+                + 'it does not match any instances',
+                exprFromRule(nearRule));
+        }
+
         // Eliminate "far" servers from candidacy.
         nearRule.remainingVms = nearRule.vms.filter(function (vm) {
             // Provisioning VMs might not have a server_uuid.
-- 
2.21.0

